drivers/virtualbox/virtualbox.go

```diff
@@ -363,6 +363,14 @@ func (d *Driver) Create() error {
 		return err
 	}
 
+	cmd, err = d.GetSSHCommand(fmt.Sprintf("sudo hostname %s", d.MachineName))
```

Comment:
You're right, it won't persist reboots. However one needs to update `/usr/local/etc/hostname` for it to persist on a b2d image.